vertx-mail-client/src/test/java/io/vertx/ext/mail/MailAttachmentTest.java

```diff
@@ -85,7 +97,7 @@ public void testConstructorFromJsonEmpty() {
 
   @Test
   public void testConstructorFromJson() {
-    final String jsonString = "{\"data\":\"YXNkZmc=\",\"name\":\"filename.jpg\"}";
+    final String jsonString = "{\"data\":\"YXNkZmc=\",\"name\":\"filename.jpg\",\"headers\":{\"Header\":[\"Value\"]}}";
```

Comment:
I guess, for compatibility reason, you should also check the the contentId is passed in the headers too.